contracts/signature_mint/SignatureMint.sol

```diff
@@ -0,0 +1,276 @@
+// SPDX-License-Identifier: Apache-2.0
+pragma solidity ^0.8.0;
+
+// Interface
+import { ISignatureMint } from "./ISignatureMint.sol";
+
+// Token
+import "@openzeppelin/contracts/token/ERC721/extensions/ERC721Enumerable.sol";
+
+// Protocol control center.
+import { ProtocolControl } from "../ProtocolControl.sol";
+
+// Signature utils
+import "@openzeppelin/contracts/utils/cryptography/ECDSA.sol";
+import "@openzeppelin/contracts/utils/cryptography/draft-EIP712.sol";
+
+// Access Control + security
+import "@openzeppelin/contracts/access/AccessControlEnumerable.sol";
+import "@openzeppelin/contracts/security/ReentrancyGuard.sol";
+
+// Royalties
+import "@openzeppelin/contracts/interfaces/IERC2981.sol";
+
+// Meta transactions
+import "@openzeppelin/contracts/metatx/ERC2771Context.sol";
+
+// Utils
+import "@openzeppelin/contracts/utils/Strings.sol";
+import "@openzeppelin/contracts/utils/Multicall.sol";
+
+contract SignatureMint is
+    
+    ISignatureMint,
+    ERC721Enumerable,
+    EIP712,
+    AccessControlEnumerable,
+    ERC2771Context,
+    IERC2981,
+    ReentrancyGuard,
+    Multicall
+
+{
+    using ECDSA for bytes32;
+    using Strings for uint256;
+
+    bytes32 private constant TYPEHASH =
+        keccak256("MintRequest(address to,string baseURI,uint256 amountToMint,uint256 validityStartTimestamp,uint256 validityEndTimestamp,bytes uid)");
```

Comment:
do TYPEHASH generally include function variable name or only variable types?

Reply:
I've been following the `Forwarder` contract for things like these. (See [TYPEHASH in `Forwarder`](https://github.com/nftlabs/nftlabs-protocols/blob/main/contracts/Forwarder.sol#L22-L23))